torch/jit/__init__.py

```diff
@@ -543,6 +517,19 @@ def _flatten(obj, params=tuple()):
     return obj_vars, obj_struct
 
 
+def _clone_inputs(args):
+    def clone_input(a):
+        if isinstance(a, Variable):
+            v = Variable(a.data.clone(), requires_grad=a.requires_grad, volatile=a.volatile)
+            if a.grad is not None:
+                v.grad = a.grad.clone()
```

Comment:
I'd rewrap the grad as well (+ can you even assign to `.grad`? I think you have to use `._grad` for this). If someone uses higher order grads, `.grad` is attached to a graph, and `clone()` will extend it

Reply:
The assignment is fine:

```
  {"_grad", (getter)THPVariable_get_grad, (setter)THPVariable_set_grad, NULL, NULL}, // only for legacy reasons 
  {"grad", (getter)THPVariable_get_grad, (setter)THPVariable_set_grad, NULL, NULL},
```

But I'll recursively call `clone_input` to handle `v.grad`.